include/grpc++/impl/codegen/client_context.h

```diff
@@ -307,6 +307,12 @@ class ClientContext {
   };
   static void SetGlobalCallbacks(GlobalCallbacks* callbacks);
 
+  // Should be used for framework-level extensions only.
+  // Applications never need to call this method.
+  grpc_call* c_call() const {
```

Comment:
Out of curiosity, why is this needed?

This seems reasonable to me, but I'd also like @ctiller to comment, since this is part of the public API.